ingestion-beam/src/main/java/com/mozilla/telemetry/util/KeyStore.java

```diff
@@ -49,6 +49,11 @@ public static KeyStore of(String metadataLocation, boolean kmsEnabled) {
   private KeyStore(String metadataLocation, boolean kmsEnabled) {
     this.metadataLocation = metadataLocation;
     this.kmsEnabled = kmsEnabled;
+
+    // throw immediately if the keyStore is misconfigured
+    if (metadataLocation == null) {
+      throw new IllegalArgumentException("Metadata location is missing.");
+    }
```

Comment:
I guess this check could be done before the assignments, but this way is also fine.